framework/src/multiapps/CentroidMultiapp.C

```diff
@@ -0,0 +1,62 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+// MOOSE includes
+#include "CentroidMultiApp.h"
+#include "MooseMesh.h"
+#include "FEProblem.h"
+
+// libMesh includes
+#include "libmesh/parallel_algebra.h"
+
+template <>
+InputParameters
+validParams<CentroidMultiApp>()
+{
+  InputParameters params = validParams<TransientMultiApp>();
+
+  params += validParams<BlockRestrictable>();
+
+  params.suppressParameter<std::vector<Point>>("positions");
+  params.suppressParameter<std::vector<FileName>>("positions_file");
+
+  return params;
+}
+
+CentroidMultiApp::CentroidMultiApp(const InputParameters & parameters)
+  : TransientMultiApp(parameters), BlockRestrictable(this)
+{
+}
+
+void
+CentroidMultiApp::fillPositions()
+{
+  MooseMesh & master_mesh = _fe_problem.mesh();
+
+  auto & bids = blockIDs();
+
+  // Detect if it can be on any block
+  bool any_bid = bids.find(Moose::ANY_BLOCK_ID) != bids.end();
```

Comment:
I'm pretty sure you don't have to do this. We shouldn't need any special logic to deal with the any case. @aeslaughter, can you confirm?

Reply:
If I remember correct this should not be needed, when BlockRestrictable is created it replaces ANY_BLOCK_ID with all the subdomains from the mesh, so it shouldn't show up in the output of `blockIDs`. Unless it has changed.

Reply:
I just looked at BlockRestrictable and it looks like ANY_BLOCK_ID is used, but there is a function 'blockRestricted` that can be called instead of doing this manually.

Reply:
Also, you can just call [`hasBlocks`](https://github.com/idaholab/moose/blob/devel/framework/src/base/BlockRestrictable.C#L196) on line 54 and avoid all of this.